libraries/AP_InertialSensor/AP_InertialSensor_MPU6000.h

```diff
@@ -18,7 +18,11 @@
 #define MPU6000_DEBUG 0
 
 #define MPU6000_SAMPLE_SIZE 14
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BH
+#define MPU6000_MAX_FIFO_SAMPLES 84
```

Comment:
@lucasdemarchi

The FIFO register of MPU60x0 is 1024 bytes, when I was modifying the code, one set of sample was 12 bytes, so the FIFO register could contain 85 samples at most, then I chose the number 84.

However, seems in the latest code, one set of sample size becomes 14 bytes, that means 73 samples at most.

The reason I set this value for BH is because we found in AP_InertialSensor_MPU6000.cpp, around line 382, if the sample number is more than MPU6000_MAX_FIFO_SAMPLES, the FIFO will be reset. During our field test, the FIFO was reset frequently like 3 out of 5. As a result the real time attitude is quite a mess due to too much data loss.

The original value is 3, as for BH, we should set this value as 5 at least. In actual fly, the remaining samples in FIFO would never pile up to 84, most of the time is 4, and 5 is the highest value I saw.